libtransmission/session.c

```diff
@@ -2359,6 +2366,55 @@ int tr_sessionGetCacheLimit_MB(tr_session const* session)
 ****
 ***/
 
+void tr_sessionSetDefaultTrackers(tr_session* session, char const* defaultTrackersStr)
+{
+    void* tmp;
+    char const* walk;
+
+    TR_ASSERT(tr_isSession(session));
+
+    /* keep the string */
+    tmp = session->defaultTrackersStr;
+    session->defaultTrackersStr = tr_strdup(defaultTrackersStr);
+    tr_free(tmp);
+
+    /* clear out the old list entries */
+    while ((tmp = tr_list_pop_front(&session->defaultTrackers)))
+    {
+        tr_free(tmp);
+    }
+
+    /* build the new list entries */
+    for (walk=defaultTrackersStr; walk && *walk;)
+    {
+        char const* delimiters = " ,;\r\n\t";
+        size_t const len = strcspn(walk, delimiters);
+        if (len)
+        {
+            char* token = tr_strndup(walk, len);
+            tr_list_append(&session->defaultTrackers, token);
+        }
+
+        if (walk[len]=='\0')
+        {
+            break;
+        }
+
+        walk += len + 1;
+    }
```

Comment:
There's a similar block of code in `tr_rpcSetList()` and I like this implementation better.

I wonder if it would make sense to move this code into `utils.[ch]` as a `tr_strsplit(const char* str, const char* delimeters, tr_list** list)` where it could be use by both this code and `tr_rpcSetList()`?